### ISSUE TITLE
com.sun.xml.ws.model.Utils uses unsupported reflection

### ISSUE THREAD

**Issue by bkabelka** [eclipse-ee4j/metro-jax-ws#54]
With JAX-WS 2.3.2, calling ``javax.xml.ws.Service.getPort(QName, Class<T>)`` (e.g. from a service implementation) leads to a ``java.lang.reflect.InaccessibleObjectException``. This exception is thrown in ``com.sun.xml.ws.model.Utils``, as the "module com.sun.xml.bind does not "opens com.sun.xml.bind.v2.model.nav" to module com.sun.xml.ws.jaxws":

```
java.lang.ExceptionInInitializerError
	at com.sun.xml.ws.jaxws/com.sun.xml.ws.model.WrapperBeanGenerator.<clinit>(WrapperBeanGenerator.java:47)
	at com.sun.xml.ws.jaxws/com.sun.xml.ws.model.RuntimeModeler.getRequestWrapperClass(RuntimeModeler.java:345)
	at com.sun.xml.ws.jaxws/com.sun.xml.ws.model.RuntimeModeler.processDocWrappedMethod(RuntimeModeler.java:771)
	at com.sun.xml.ws.jaxws/com.sun.xml.ws.model.RuntimeModeler.processMethod(RuntimeModeler.java:701)
	at com.sun.xml.ws.jaxws/com.sun.xml.ws.model.RuntimeModeler.processClass(RuntimeModeler.java:461)
	at com.sun.xml.ws.jaxws/com.sun.xml.ws.model.RuntimeModeler.buildRuntimeModel(RuntimeModeler.java:290)
	at com.sun.xml.ws.jaxws/com.sun.xml.ws.db.DatabindingImpl.<init>(DatabindingImpl.java:70)
	at com.sun.xml.ws.jaxws/com.sun.xml.ws.db.DatabindingProviderImpl.create(DatabindingProviderImpl.java:44)
	at com.sun.xml.ws.jaxws/com.sun.xml.ws.db.DatabindingProviderImpl.create(DatabindingProviderImpl.java:28)
	at com.sun.xml.ws.jaxws/com.sun.xml.ws.db.DatabindingFactoryImpl.createRuntime(DatabindingFactoryImpl.java:90)
	at com.sun.xml.ws.jaxws/com.sun.xml.ws.client.WSServiceDelegate.buildRuntimeModel(WSServiceDelegate.java:852)
	at com.sun.xml.ws.jaxws/com.sun.xml.ws.client.WSServiceDelegate.createSEIPortInfo(WSServiceDelegate.java:869)
	at com.sun.xml.ws.jaxws/com.sun.xml.ws.client.WSServiceDelegate.addSEI(WSServiceDelegate.java:832)
	at com.sun.xml.ws.jaxws/com.sun.xml.ws.client.WSServiceDelegate.getPort(WSServiceDelegate.java:421)
	at com.sun.xml.ws.jaxws/com.sun.xml.ws.client.WSServiceDelegate.getPort(WSServiceDelegate.java:390)
	at com.sun.xml.ws.jaxws/com.sun.xml.ws.client.WSServiceDelegate.getPort(WSServiceDelegate.java:372)
	at java.xml.ws/javax.xml.ws.Service.getPort(Service.java:139)
	[...]
Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make static com.sun.xml.bind.v2.model.nav.ReflectionNavigator com.sun.xml.bind.v2.model.nav.ReflectionNavigator.getInstance() accessible: module com.sun.xml.bind does not "opens com.sun.xml.bind.v2.model.nav" to module com.sun.xml.ws.jaxws
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:340)
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:280)
	at java.base/java.lang.reflect.Method.checkCanSetAccessible(Method.java:198)
	at java.base/java.lang.reflect.Method.setAccessible(Method.java:192)
	at com.sun.xml.ws.jaxws/com.sun.xml.ws.model.Utils$1.run(Utils.java:51)
	at com.sun.xml.ws.jaxws/com.sun.xml.ws.model.Utils$1.run(Utils.java:46)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at com.sun.xml.ws.jaxws/com.sun.xml.ws.model.Utils.<clinit>(Utils.java:45)
	... 58 more
``` 

A possible workaround is to start your application with the ``--add-opens com.sun.xml.bind/com.sun.xml.bind.v2.model.nav=com.sun.xml.ws.jaxws`` flag. However, I would expect the JAX-WS library to work without any additional ``--add-opens`` statements.

**Comment by smillidge** [comment]
Can you create a PR for this as it looks like you understand the fix?

**Comment by bkabelka** [comment]
Well, I am not familiar with the code. However, if opening com.sun.xml.bind/com.sun.xml.bind.v2.model.nav to com.sun.xml.ws.jaxws is fine, I can open a PR in jaxb-ri for that.

**Comment by lukasj** [comment]
this should be fixed by jaxb-ri 2.3.3-b02 and adopted in current master

**Comment by Deepak00-Sh** [comment]
I'm getting this :
to start bean 'connectorFactory'; nested exception is java.lang.ExceptionInInitializerError
2024-04-04 07:50:11,385 [main] ERROR   ContextLoader        Context initialization failed
org.springframework.context.ApplicationContextException: Failed to start bean 'connectorFactory'; nested exception is java.lang.ExceptionInInitializerError
        at org.springframework.context.support.DefaultLifecycleProcessor.doStart(DefaultLifecycleProcessor.java:181)
        at org.springframework.context.support.DefaultLifecycleProcessor.access$200(DefaultLifecycleProcessor.java:54)
        at org.springframework.context.support.DefaultLifecycleProcessor$LifecycleGroup.start(DefaultLifecycleProcessor.java:356)
        at java.lang.Iterable.forEach(Iterable.java:75)
        at org.springframework.context.support.DefaultLifecycleProcessor.startBeans(DefaultLifecycleProcessor.java:155)
        at org.springframework.context.support.DefaultLifecycleProcessor.onRefresh(DefaultLifecycleProcessor.java:123)
        at org.springframework.context.support.AbstractApplicationContext.finishRefresh(AbstractApplicationContext.java:937)
        at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:586)
        at org.springframework.web.context.ContextLoader.configureAndRefreshWebApplicationContext(ContextLoader.java:399)
        at org.springframework.web.context.ContextLoader.initWebApplicationContext(ContextLoader.java:278)
        at org.springframework.web.context.ContextLoaderListener.contextInitialized(ContextLoaderListener.java:103)
        at org.apache.catalina.core.StandardContext.listenerStart(StandardContext.java:4462)
        at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:4914)
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:171)
        at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:683)
        at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:658)
        at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:661)
        at org.apache.catalina.startup.HostConfig.deployWAR(HostConfig.java:1025)
        at org.apache.catalina.startup.HostConfig$DeployWar.run(HostConfig.java:1919)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75)
        at java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:112)
        at org.apache.catalina.startup.HostConfig.deployWARs(HostConfig.java:826)
        at org.apache.catalina.startup.HostConfig.deployApps(HostConfig.java:476)
        at org.apache.catalina.startup.HostConfig.start(HostConfig.java:1619)
        at org.apache.catalina.startup.HostConfig.lifecycleEvent(HostConfig.java:318)
        at org.apache.catalina.util.LifecycleBase.fireLifecycleEvent(LifecycleBase.java:114)
        at org.apache.catalina.util.LifecycleBase.setStateInternal(LifecycleBase.java:402)
        at org.apache.catalina.util.LifecycleBase.setState(LifecycleBase.java:345)
        at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:893)
        at org.apache.catalina.core.StandardHost.startInternal(StandardHost.java:794)
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:171)
        at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1332)
        at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1322)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75)
        at java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:134)
        at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:866)
        at org.apache.catalina.core.StandardEngine.startInternal(StandardEngine.java:248)
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:171)
        at org.apache.catalina.core.StandardService.startInternal(StandardService.java:433)
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:171)
        at org.apache.catalina.core.StandardServer.startInternal(StandardServer.java:921)
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:171)
        at org.apache.catalina.startup.Catalina.start(Catalina.java:772)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:347)
        at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:478)
Caused by: java.lang.ExceptionInInitializerError: null
        at com.sun.xml.ws.model.WrapperBeanGenerator.<clinit>(WrapperBeanGenerator.java:47)
        at com.sun.xml.ws.model.RuntimeModeler.getExceptionBeanClass(RuntimeModeler.java:371)
        at com.sun.xml.ws.model.RuntimeModeler.processExceptions(RuntimeModeler.java:1224)
        at com.sun.xml.ws.model.RuntimeModeler.processDocWrappedMethod(RuntimeModeler.java:939)
        at com.sun.xml.ws.model.RuntimeModeler.processMethod(RuntimeModeler.java:701)
        at com.sun.xml.ws.model.RuntimeModeler.processClass(RuntimeModeler.java:461)
        at com.sun.xml.ws.model.RuntimeModeler.buildRuntimeModel(RuntimeModeler.java:290)
        at com.sun.xml.ws.db.DatabindingImpl.<init>(DatabindingImpl.java:70)
        at com.sun.xml.ws.db.DatabindingProviderImpl.create(DatabindingProviderImpl.java:44)
        at com.sun.xml.ws.db.DatabindingProviderImpl.create(DatabindingProviderImpl.java:28)
        at com.sun.xml.ws.db.DatabindingFactoryImpl.createRuntime(DatabindingFactoryImpl.java:90)
        at com.sun.xml.ws.client.WSServiceDelegate.buildRuntimeModel(WSServiceDelegate.java:852)
        at com.sun.xml.ws.client.WSServiceDelegate.createSEIPortInfo(WSServiceDelegate.java:869)
        at com.sun.xml.ws.client.WSServiceDelegate.addSEI(WSServiceDelegate.java:832)
        at com.sun.xml.ws.client.WSServiceDelegate.getPort(WSServiceDelegate.java:421)
        at com.sun.xml.ws.client.WSServiceDelegate.getPort(WSServiceDelegate.java:390)
        at com.sun.xml.ws.client.WSServiceDelegate.getPort(WSServiceDelegate.java:372)
        at javax.xml.ws.Service.getPort(Service.java:119)
        at 
Caused by: java.lang.IllegalStateException: ReflectionNavigator.getInstance can't be found
        at com.sun.xml.ws.model.Utils$1.run(Utils.java:54)
        at com.sun.xml.ws.model.Utils$1.run(Utils.java:46)
        at java.security.AccessController.doPrivileged(Native Method)
        at com.sun.xml.ws.model.Utils.<clinit>(Utils.java:45)
        ... 76 common frames omitted

Could you please guide me to fix this ?
